doc/api-details.md

```diff
@@ -164,6 +176,61 @@ $ curl -X 'PUT' \
 {"_id":"61bda8f2eb1a63d2b7152418","kind":"node","name":"checkout-test","revision":{"tree":"mainline","url":"https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git","branch":"master","commit":"2a987e65025e2b79c6d453b78cb5985ac6e5eb26","describe":"v5.16-rc4-31-g2a987e65025e"},"parent":null,"status":"pending","result":null, "created":"2022-02-02T11:23:03.157648", "updated":"2022-02-02T12:23:03.157648"}
 ```
 
+### State diagram
+
+The Node objects are governed by the following state machine:
+
+```mermaid
+graph LR
+  running(Running) --> job{successful?}
+  running -.-> |timeout| done(Done)
+  job --> |no| done(Done)
+  job --> |yes| ready(Ready)
+  ready -.-> |timeout| done
+  ready --> |holdoff expired| pending(Pending)
+  pending -.-> |timeout| done
+  pending --> |child nodes done| done
+```
+
+The state of the Node is kept in the `state` field in the `Node` models.  The
+different state values are described below:
+
+Running
+: The job has been scheduled and there is no result yet.
+
+Ready
+: The job was successful.  Child nodes that depend on the job's success can now
+  be created.
+
+Pending
+: The holdoff time has been reached, the node is now waiting for any pending
+  child nodes to reach the Done state.
+
+Done
+: The node has reached its final state.
```

Comment:
I would suggest using Bullet List here.

Reply:
This is actually the Markdown syntax for definitions

Reply:
Oh, okay. I was unaware of that.